data/src/main/java/com/microsoft/azure/kusto/data/UriUtils.java

```diff
@@ -10,14 +11,16 @@ private UriUtils() {
     }
 
     public static String setPathForUri(String uri, String path, boolean ensureTrailingSlash) throws URISyntaxException {
-        if (ensureTrailingSlash && !path.endsWith("/")) {
-            path += "/";
-        }
-        if (!path.startsWith("/")) {
-            path = "/" + path;
+        if (ensureTrailingSlash) {
+            path = StringUtils.appendIfMissing(path, "/");
```

Comment:
What's the purpose of this block? It seems superfluous to me.

Reply:
removed